include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -505,8 +515,10 @@ struct __get_sycl_range
         auto __n = __last - __first;
         assert(__n > 0);
 
-        auto res_src =
-            __process_input_iter<_LocalAccMode>(__first.base(), oneapi::dpl::end(__first.base().get_buffer()));
+        auto __base_iter = __first.base();
```

Comment:
I think that new 4 lines are equivalent to the couple of line that we had before... No?

`__process_input_iter<_LocalAccMode>(__first.base(), oneapi::dpl::end(__first.base().get_buffer()));`

Reply:
It is, but only if `first.base()` is a `sycl_iterator`.  If `first.base()` is instead a `dpct::device_pointer` this fails because no overload exists with such combination of types for `__process_input_iter` (`dpct::device_pointer`,`sycl_iterator`).

Reply:
I see..